src/OpenColorIO/GpuShaderClassWrapper.cpp

```diff
@@ -38,20 +45,28 @@ std::string MetalShaderClassWrapper::generateClassWrapperHeader(GpuShaderText& k
     for(const auto& param : m_functionParameters)
     {
         size_t openAngledBracketPos = param.name.find('[');
-        bool isArray = openAngledBracketPos != std::string::npos;
-        if(!isArray)
+        if(!param.isArray)
             kw.newLine()    << "this->" << param.name  << " = " << param.name  << ";";
         else
         {
             size_t closeAngledBracketPos = param.name.find(']');
             std::string variableName = param.name.substr(0, openAngledBracketPos);
             
             kw.newLine()    << "for(int i = 0; i < "
-                            << param.name.substr(openAngledBracketPos+1, closeAngledBracketPos-openAngledBracketPos-1)
+                            << GetArrayLengthVariableName(variableName)
```

Comment:
The array size specifies passed data size upper limit, but in practice what we pass as constant/uniform buffer contains less data. To generate entry function and constant buffer definition (in msl.mm), this upper limit needs to be known but information is not available and to make it available, I either have to parse the shader and extract max array size or add max length member to `VectorFloat` and `VectorInt`. First one is very dirty, and second one is clean but touches OpenColorIO.h. Therefore, I fixed the problem by passing how many elements of this array is passed through constant/uniform buffer. 

I actually like this approach because this way the size of data needs to be passed to GPU is reduced.